src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/Builder/ControllerEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,179 @@
+        /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with controller actions.</returns>
+        public static IEndpointConventionBuilder MapControllers(this IEndpointRouteBuilder routes)
+        {
+            if (routes == null)
+            {
+                throw new ArgumentNullException(nameof(routes));
+            }
+
+            EnsureControllerServices(routes);
+
+            return GetOrCreateDataSource(routes);
+        }
+
+        /// <summary>
+        /// Adds endpoints for controller actions to the <see cref="IEndpointRouteBuilder"/> and adds the default route
+        /// <c>{controller=Home}/{action=Index}/{id?}</c>.
+        /// </summary>
+        /// <param name="routes">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with controller actions.</returns>
+        public static IEndpointConventionBuilder MapDefaultControllerRoute(this IEndpointRouteBuilder routes)
+        {
+            if (routes == null)
+            {
+                throw new ArgumentNullException(nameof(routes));
+            }
+
+            EnsureControllerServices(routes);
+
+            var dataSource = GetOrCreateDataSource(routes);
+            dataSource.AddRoute(new ConventionalRouteEntry(
+                "default",
+                "{controller=Home}/{action=Index}/{id?}",
+                defaults: null,
+                constraints: null,
+                dataTokens: null));
+
+            return dataSource;
+        }
+
+        /// <summary>
+        /// Adds endpoints for controller actions to the <see cref="IEndpointRouteBuilder"/> and specifies a route
+        /// with the given <paramref name="name"/>, <paramref name="template"/>, 
+        /// <paramref name="defaults"/>, <paramref name="constraints"/>, and <paramref name="dataTokens"/>.
+        /// </summary>
+        /// <param name="routes">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="name">The name of the route.</param>
+        /// <param name="template">The URL pattern of the route.</param>
+        /// <param name="defaults">
+        /// An object that contains default values for route parameters. The object's properties represent the
+        /// names and values of the default values.
+        /// </param>
+        /// <param name="constraints">
+        /// An object that contains constraints for the route. The object's properties represent the names and
+        /// values of the constraints.
+        /// </param>
+        /// <param name="dataTokens">
+        /// An object that contains data tokens for the route. The object's properties represent the names and
+        /// values of the data tokens.
+        /// </param>
+        public static void MapControllerRoute(
+            this IEndpointRouteBuilder routes,
+            string name,
+            string template,
```

Comment:
Is there a reason for template over pattern? Is it to making it easier to copy over existing code?

Reply:
No not really. This is template for consistency with a bunch of older stuff.  It's nice if you can find and replace your way to victory. If you think of a good reason to change it I'm all ears.